Applications/Spire/Include/Spire/Spire/AnyValueModel.hpp

```diff
@@ -0,0 +1,49 @@
+#ifndef SPIRE_ANY_VALUE_MODEL_HPP
+#define SPIRE_ANY_VALUE_MODEL_HPP
+#include <memory>
+#include "Spire/Spire/AnyRef.hpp"
+#include "Spire/Spire/Spire.hpp"
+#include "Spire/Spire/ValueModel.hpp"
+
+namespace Spire {
+
+  /**
+   * Implements a ValueModel by wrapping an existing ValueModel with a type-safe
+   * generic interface.
+   */
+  class AnyValueModel : public ValueModel<AnyRef> {
+    public:
+      using Type = ValueModel<AnyRef>::Type;
+
+      using UpdateSignal = ValueModel<AnyRef>::UpdateSignal;
+
+      /**
+       * Constructs an AnyValueModel over an existing ValueModel.
+       * @param source The ValueModel to wrap.
+       */
+      template<typename T>
```

Comment:
Should these use Type instead of T?

Reply:
`Type` and `T` are different. `Type` of the AnyValueModel is `AnyRef`, while `T` used in the constructor could be of any type. Taking the unit test in AnyValueModelTester for example, T is `int`.